### PR TITLE
Add pr-build-connector-template

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: .github/workflows/pull-request-build-connector-template.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+jobs:
+    ubuntu-build:
+        name: Build on Ubuntu
+        runs-on: ubuntu-latest
+        concurrency:
+          group: ${{ github.head_ref }}-ubuntu-build
+          cancel-in-progress: true
+        steps:
+            -   name: Checkout Repository
+                uses: actions/checkout@v1
```
Comment (ThisaruGuruge) [inline .github/workflows/pull-request-build-connector-template.yml:25]: Let's use `actions/checkout@v3` here. 
```suggestion
                uses: actions/checkout@v3
```

File: .github/workflows/pull-request-build-connector-template.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            -   name: Build the Package
+                env:
+                    packageUser: ${{ github.actor }}
+                    packagePAT: ${{ secrets.GITHUB_TOKEN }}
+                run: |
+                  ./gradlew build -x test ${{ inputs.additional-build-flags }}
+                  ./gradlew test ${{ inputs.additional-test-flags }}
+
+            -   name: Generate Codecov Report
+                uses: codecov/codecov-action@v2
```
Comment (ThisaruGuruge) [inline .github/workflows/pull-request-build-connector-template.yml:51]: I think we need to use codecov-action v3 as well. Please check and update.

Comment (MohamedSabthar) [reply .github/workflows/pull-request-build-connector-template.yml:51]: Updated with [f91e8c9](https://github.com/ballerina-platform/ballerina-standard-library/pull/4838/commits/f91e8c976848391ac77970102c5a830d415094e8)

File: .github/workflows/pr-build-connector-template.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        default: ""
+      additional-test-flags:
+        required: false
+        type: string
+        default: ""
+
+
+jobs:
+    ubuntu-build:
+        name: Build on Ubuntu
```
Comment (ThisaruGuruge) [inline .github/workflows/pr-build-connector-template.yml:18]: We can change the job to `Build` since we don't have separate builds with this.

Comment (MohamedSabthar) [reply .github/workflows/pr-build-connector-template.yml]: updated

File: .github/workflows/pr-build-connector-template.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        type: string
+        default: ""
+      additional-test-flags:
+        required: false
+        type: string
+        default: ""
+
+
+jobs:
+    ubuntu-build:
```
Comment (ThisaruGuruge) [inline .github/workflows/pr-build-connector-template.yml:17]:
```suggestion
    build:
```

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**